dashboard/app/assets/javascripts/dialog.js.erb

```diff
@@ -81,3 +113,49 @@ Dialog.prototype.hide = function() {
   $(this.div).modal('hide');
 };
 
+/**
+ * This processes optional hideOptions that were provided to show().
+ * At the moment it will play an animation of the dialog moving and resizing to
+ * the location and dimensions of a specified div, while also fading out.
+ * Certain elements are faded out more quickly so that they are gone before
+ * the dialog gets too small.
+ */
+Dialog.prototype.processHideOptions = function(options) {
+
+  var startCss = {};
+  startCss.opacity = '1';
+  startCss.left    = this.div.css('left');
+  startCss.top     = this.div.css('top');
+  startCss.width   = this.div.css('width');
+  startCss.height  = this.div.css('height');
+
+  var endCss = {};
+  endCss.opacity  = '0';
+  endCss.overflow = "visible";
+  endCss.left     = $(options.endTarget).offset().left + 320;
+  endCss.top      = $(options.endTarget).offset().top - $(window).scrollTop();
+  endCss.width    = $(options.endTarget).css("width");
+  endCss.height   = $(options.endTarget).css("height");
+
+  // Let's also fade the background out.
+  $(".modal-backdrop").animate({opacity: 0}, 1000);
+
+  // And a bunch of other elements
+  $(".farSide").animate({opacity: 0}, 300);
+  $("#x-close").animate({opacity: 0}, 300);
+  $(".dialog-title").animate({opacity: 0}, 300);
+  $(".modal-content p").animate({"font-size": "13px"}, 1000)
```

Comment:
Will all the `300` usages typically be tweaked together? Could be handy to extract variables for the 300/1000